src/lightning_lite/utilities/data.py

```diff
@@ -0,0 +1,424 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
Many of the functions in this file are needed in Lite, but have PL specific implementations. This is why I needed to copy them as a whole and delete PL specific code.

Reply:
we can leave it for later :)